drafts/bitcartel-nu0-tx-format/draft.rst

```diff
@@ -0,0 +1,207 @@
+
+Specification
+=============
+
+Zcash launched with support for upstream Bitcoin version 1 transactions and a custom version 2 transaction format which added fields required for shielded transactions.
+
+======== =============== =========================== =======
+Version  Field           Description                 Type
+======== =============== =========================== =======
+>= 1     version         positive value              int32
+>= 1     in_count        varint                      1-9 bytes
+>= 1     tx_inputs       list of inputs              vector
+>= 1     out_count       varint                      1-9 bytes
+>= 1     tx_outputs      list of outputs             vector
+>= 1     lock_time       block height or timestamp   uint32
+>= 2     nJoinSplit      varint                      1-9 bytes
+>= 2     vJoinSplit      list of joinsplits          vector
+>= 2     joinSplitPubKey joinSplitSig public key     32 bytes
+>= 2     joinSplitSig    signature                   64 bytes
+======== =============== =========================== =======
+
+Version 1 and 2 transaction formats are legacy formats.
+
+Version 3 transaction format will be introduced for Overwinter.
+
+The version field must have its most significant bit set.
+
+There will be new fields for:
+
+* branch id
+* expiry height
+
+======== =============== =========================== =======
+Version  Field           Description                 Type
+======== =============== =========================== =======
+>= 3     version         must have MSB set           int32
+>= 3     branch_id       branch/fork identifier      uint32
+>= 3     expiry_height   block height                uint32
+>= 1     in_count        varint                      1-9 bytes
+>= 1     tx_inputs       list of inputs              vector
+>= 1     out_count       varint                      1-9 bytes
+>= 1     tx_outputs      list of outputs             vector
+>= 1     lock_time       block height or timestamp   uint32
+>= 2     nJoinSplit      varint                      1-9 bytes
+>= 2     vJoinSplit      list of joinsplits          vector
+>= 2     joinSplitPubKey joinSplitSig public key     32 bytes
+>= 2     joinSplitSig    signature                   64 bytes
+======== =============== =========================== =======
+
+
+Transaction Version
+-------------------
+
+The version field is always serialized in little-endian format.
+
+Version 1 transaction 5c6ba844e1ca1c8083cd53e29971bd82f1f9eea1f86c1763a22dd4ca183ae061
+
+* begins with 0x01000000
+* 32-bit signed integer 00 00 00 01 == 1
+
+Version 2 transaction 4435bf8064e74f01262cb1725fd9b53e600fa285950163fd961bed3a64260d8b
+
+* begins with 0x02000000
+* 32-bit signed integer 00 00 00 02 == 2
+
+Legacy parsers require the transaction version number to be positive.
+
+Overwinter parsers also require the transaction version number to be positive, however the serialized version field must have the most significant bit set.
```

Comment:
If we separate the ``control`` and ``nVersion`` fields then edit this accordingly.